src/main.rs

```diff
@@ -81,8 +81,17 @@ fn main() -> Result<(), Box<dyn Error>> {
         );
     let matches = match opts.parse(std::env::args()) {
         Ok(it) => it,
+        Err(getopts::Fail::UnrecognizedOption(string)) => {
+            println!(
+                "{}",
+                opts.usage(&format!(
+                    "Unknown option '{string}'! Please rerun with one of these:"
```

Comment:
"One of these" or rather "some of these"? :wrench:

Reply:
Hmm, how it currently is "some of these", since there are multiple options, which are actually not required.

It is also possible, that the user actually supplied a required option (like `--stdin`) but this usage message still prints, since this triggers on any unrecognized option in the argument list.

Maybe another cli parser, that supports subcommands might be a solution here. I've worked with `clap` in the past, but it is rather big, and there are probably better fitting solutions to this. Might be at least more clear then "radio button"-like options, where one out of three is required.

Reply:
I just checked with the programs I have locally installed, and most just output "Unkown option: {string}" followed by the usage output. So maybe something like that? Or do you have examples of programs, that behave differently?

Reply:
Basically, the message should also be written to `stderr` instead of `stdout`.  I will fix this together with the prompted message in a single commit.

Reply:
:wrench: :heavy_check_mark: